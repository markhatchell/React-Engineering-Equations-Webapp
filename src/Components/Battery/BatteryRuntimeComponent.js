import React from 'react';
import BaseEquation from '../BaseEquation';
import '../../Styles/Equations.css';
const Battery = require('js-electrical-engineering-equations').Battery;
const getFloat = require('js-electrical-engineering-equations').getFloat;

class BatteryRuntimeComponent extends BaseEquation {

  constructor(props) {
    if (props.hasOwnProperty('onChangeTitle')) {
      props.onChangeTitle('Battery Runtime');
    }
    super(props);
    this.state = {
      batteryCapacityInAmpHours: 1,
      currentDrawInAmps: 1,
      efficiency: .8,
      battery: new Battery()
    }
  }

  componentDidMount() {
    this.calculate();
  }

  render() {
    return (
      <div className="equationBox">
        <h3>Battery Runtime</h3>
        <div className="equationBoxInner">
          <p>Calculate the runtime on a battery. This equation does not take into account the diminished runtime effect of
            high current applications.</p>
          <p>The efficiency of the system is set to 80% by default.</p>
          <div className="tableContainer">
            <form onSubmit={(e) => this.handleSubmit(e)}>
              <table className="table" width="100%">
                <tbody>
                <tr>
                  <td align="left">
                    <label htmlFor="batteryCapacityInAmpHours">Battery capacity in amp hours</label>
                  </td>
                  <td>
                    <input id="batteryCapacityInAmpHours" value={this.state.batteryCapacityInAmpHours}
                           autoComplete="off"
                           onChange={(e) => this.collectFloatValueFor('batteryCapacityInAmpHours', e)}/>
                  </td>
                </tr>
                <tr>
                  <td align="left">
                    <label htmlFor="currentDrawInAmps">Current draw in amps</label>
                  </td>
                  <td>
                    <input id="currentDrawInAmps" value={this.state.currentDrawInAmps}
                           autoComplete="off"
                           onChange={(e) => this.collectFloatValueFor('currentDrawInAmps', e)}/>
                  </td>
                </tr>
                <tr>
                  <td align="left">
                    <label htmlFor="efficiency">Efficiency of system</label>
                  </td>
                  <td>
                    <input id="efficiency" value={this.state.efficiency}
                           autoComplete="off"
                           onChange={(e) => this.collectFloatValueFor('efficiency', e)}/>
                  </td>
                </tr>
                <tr>
                  <td align="left">
                    Battery capacity in amp hours
                  </td>
                  <td align="left">
                    {this.state.battery.getBatteryCapacityInAmpHours()}
                  </td>
                </tr>
                <tr>
                  <td align="left">
                    Battery capacity in coulombs
                  </td>
                  <td align="left">
                    {this.state.battery.getBatteryCapacityInCoulombs()}
                  </td>
                </tr>
                <tr>
                  <td align="left">
                    Battery run time in min
                  </td>
                  <td align="left">
                    {this.state.battery.getRunTimeInMin()}
                  </td>
                </tr>
                <tr>
                  <td align="left">
                    Battery run time in hours
                  </td>
                  <td align="left">
                    {this.state.battery.getRunTimeInHours()}
                  </td>
                </tr>
                <tr>
                  <td align="left">
                    Battery run time in days
                  </td>
                  <td align="left">
                    {getFloat(this.state.battery.getRunTimeInHours() / 24, 4)}
                  </td>
                </tr>
                <tr>
                  <td align="left">
                  </td>
                  <td align="right">
                    <button type="submit">Calculate</button>
                  </td>
                </tr>
                </tbody>
              </table>
            </form>
          </div>
        </div>
      </div>
    );
  }

  updateBattery() {
    let battery = this.state.battery;
    battery.setBatteryCapacityInAmpHours(parseFloat(this.state.batteryCapacityInAmpHours));
    battery.setCurrentDraw(parseFloat(this.state.currentDrawInAmps));
    battery.efficiency = this.state.efficiency;
    battery.calculateRuntime();
    return battery;
  }

  calculate() {
    this.setState(Object.assign(
      this.state,
      {
        battery: this.updateBattery()
      }
    ));

  }

  handleSubmit(e) {
    this.calculate();
    e.preventDefault();
    return false;
  }

}


export default BatteryRuntimeComponent;