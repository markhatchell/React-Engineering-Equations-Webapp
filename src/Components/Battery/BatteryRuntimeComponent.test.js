import React from 'react';
import renderer from 'react-test-renderer';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import BatteryRuntimeComponent from "./BatteryRuntimeComponent";

configure({ adapter: new Adapter() });

const dom = renderer.create(
  <BatteryRuntimeComponent />,
);

describe('BatteryRuntimeComponent', () => {

  test('snapshot matches', () => {
    let tree = dom.toJSON();
    expect(tree).toMatchSnapshot();
  });

  let component = shallow(<BatteryRuntimeComponent />);
  let instance = component.instance();


  test('wavelength should be 1', () => {
    component.setState(Object.assign(
      {},
      component.state(),
      {
        batteryCapacityInAmpHours: 1,
        currentDrawInAmps: 1,
        efficiency: .8
      }
    ));
    instance.calculate();
    expect(component.state("battery").getRunTimeInMin()).toEqual(48);
  });

});
