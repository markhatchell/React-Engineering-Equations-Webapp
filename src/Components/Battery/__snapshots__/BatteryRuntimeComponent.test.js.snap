// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BatteryRuntimeComponent snapshot matches 1`] = `
<div
  className="MuiPaper-root-2 MuiPaper-elevation2-6 MuiPaper-rounded-3 MuiCard-root-1 equationBox"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="MuiCardContent-root-29 equationBoxInner"
    >
      <p>
        Calculate the runtime on a battery. This equation does not take into account the diminished runtime effect of high current applications.
      </p>
      <p>
        The efficiency of the system is set to 80% by default.
      </p>
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-30 MuiFormControl-marginNormal-31 MuiFormControl-fullWidth-33"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-39 MuiInputLabel-root-34 MuiInputLabel-formControl-35 MuiInputLabel-animated-38 MuiInputLabel-shrink-37"
          data-shrink={true}
          htmlFor="full-width"
        >
          Battery capacity in amp hours
        </label>
        <div
          className="MuiInput-root-44 MuiInput-fullWidth-51 MuiInput-formControl-45 MuiInput-underline-48"
        >
          <input
            aria-invalid={false}
            aria-required={false}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-52"
            defaultValue={undefined}
            disabled={false}
            id="full-width"
            name="batteryCapacityInAmpHours"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="text"
            value={1}
          />
        </div>
      </div>
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-30 MuiFormControl-marginNormal-31 MuiFormControl-fullWidth-33"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-39 MuiInputLabel-root-34 MuiInputLabel-formControl-35 MuiInputLabel-animated-38 MuiInputLabel-shrink-37"
          data-shrink={true}
          htmlFor="full-width"
        >
          Current draw in amps
        </label>
        <div
          className="MuiInput-root-44 MuiInput-fullWidth-51 MuiInput-formControl-45 MuiInput-underline-48"
        >
          <input
            aria-invalid={false}
            aria-required={false}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-52"
            defaultValue={undefined}
            disabled={false}
            id="full-width"
            name="currentDrawInAmps"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="text"
            value={1}
          />
        </div>
      </div>
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-30 MuiFormControl-marginNormal-31 MuiFormControl-fullWidth-33"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-39 MuiInputLabel-root-34 MuiInputLabel-formControl-35 MuiInputLabel-animated-38 MuiInputLabel-shrink-37"
          data-shrink={true}
          htmlFor="full-width"
        >
          Efficiency of system
        </label>
        <div
          className="MuiInput-root-44 MuiInput-fullWidth-51 MuiInput-formControl-45 MuiInput-underline-48"
        >
          <input
            aria-invalid={false}
            aria-required={false}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-52"
            defaultValue={undefined}
            disabled={false}
            id="full-width"
            name="efficiency"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="text"
            value={0.8}
          />
        </div>
      </div>
      <table
        className="MuiTable-root-57"
      >
        <tr
          className="MuiTableRow-root-58"
        >
          <td
            align="left"
            aria-sort={null}
            className="MuiTableCell-root-63"
            scope={undefined}
          >
            Battery capacity in amp hours
          </td>
          <td
            align="left"
            aria-sort={null}
            className="MuiTableCell-root-63"
            scope={undefined}
          >
            1
          </td>
        </tr>
        <tr
          className="MuiTableRow-root-58"
        >
          <td
            align="left"
            aria-sort={null}
            className="MuiTableCell-root-63"
            scope={undefined}
          >
            Battery capacity in coulombs
          </td>
          <td
            align="left"
            aria-sort={null}
            className="MuiTableCell-root-63"
            scope={undefined}
          >
            3600
          </td>
        </tr>
        <tr
          className="MuiTableRow-root-58"
        >
          <td
            align="left"
            aria-sort={null}
            className="MuiTableCell-root-63"
            scope={undefined}
          >
            Battery run time in min
          </td>
          <td
            align="left"
            aria-sort={null}
            className="MuiTableCell-root-63"
            scope={undefined}
          >
            48
          </td>
        </tr>
        <tr
          className="MuiTableRow-root-58"
        >
          <td
            align="left"
            aria-sort={null}
            className="MuiTableCell-root-63"
            scope={undefined}
          >
            Battery run time in hours
          </td>
          <td
            align="left"
            aria-sort={null}
            className="MuiTableCell-root-63"
            scope={undefined}
          >
            0.8
          </td>
        </tr>
        <tr
          className="MuiTableRow-root-58"
        >
          <td
            align="left"
            aria-sort={null}
            className="MuiTableCell-root-63"
            scope={undefined}
          >
            Battery run time in days
          </td>
          <td
            align="left"
            aria-sort={null}
            className="MuiTableCell-root-63"
            scope={undefined}
          >
            0.0333
          </td>
        </tr>
      </table>
    </div>
    <div
      className="MuiCardActions-root-71"
    >
      <button
        className="MuiButtonBase-root-88 MuiButton-root-73 MuiButton-raised-78 MuiButton-raisedPrimary-79 MuiCardActions-action-72"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "marginLeft": "auto",
          }
        }
        tabIndex="4"
        type="submit"
      >
        <span
          className="MuiButton-label-74"
        >
          Calculate
        </span>
        <span
          className="MuiTouchRipple-root-91"
        />
      </button>
    </div>
  </form>
</div>
`;
